src/interceptors/ClientRequest/agents.ts

```diff
@@ -66,15 +75,23 @@ export class MockHttpsAgent extends https.Agent {
 
   public createConnection(options: any, callback: any): net.Socket {
     const createConnection =
-      (this.customAgent instanceof https.Agent &&
-        this.customAgent.createConnection) ||
-      super.createConnection
+      this.customAgent instanceof https.Agent
+        ? this.customAgent.createConnection
+        : super.createConnection
+
+    const createConnectionOptions =
+      this.customAgent instanceof https.Agent
+        ? {
+            ...options,
+            ...this.customAgent.options,
+          }
+        : options
 
     const socket = new MockHttpSocket({
       connectionOptions: options,
       createConnection: createConnection.bind(
         this.customAgent || this,
-        options,
+        createConnectionOptions,
```

Comment:
We don't really have any direct tests for this. Is this a reason for us to add `agents.test.ts` and test these things on the unit level?

Reply:
I saw your merged it, but I'm wondering if we need here a unit test to make sure we merge the options correctly. 
I tested the normalize function, but I'm not sure if it's enough

Reply:
Ideally, we can have a follow up PR that unit-tests the agents. I think that's the right surface to do it. Does this makes sense to you?

Reply:
Yes. I opened an issue for this, so we won't forget about this:
https://github.com/mswjs/interceptors/issues/703